specification/data-faas.md

```diff
@@ -2,7 +2,7 @@
 
 This document defines how to describe an instance of a function that runs without provisioning or managing of servers (also known as serverless) with spans.
 
-Span `name` should be set to the function name being executed. Depending on the trigger of the function, additional attributes MUST be set. For example, an HTTP trigger MUST follow the [HTTP Server semantic conventions](data-http.md#http-server-semantic-conventions).
+Span `name` should be set to the function name being executed. Depending on the value of the `faas.trigger` attribute, additional attributes MUST be set. For example, an `http` trigger MUST follow the [HTTP Server semantic conventions](data-http.md#http-server-semantic-conventions).
```

Comment:
But is that really doable? I assume in many cases the HTTP span would be created by another sensor that comes before or after the faas sensor. Is the faas sensors supposed to integrate into the other sensor or is it fine if we have nested spans instead of all the information on a single span?

Reply:
In my understanding, a FaaS sensor decides the correct `faas.trigger` value and then forwards  accordingly the tagging to the correct sensor

Reply:
At least for AWS lambda NodeJs the translation from HTTP to lambda function call (incl. load balancing) happens outside of the processes hosting the function. There is no real HTTP happening inside lambda, a function is invoked because of an HTTP request and the function gets input data converted into an event format (see e.g. https://docs.aws.amazon.com/lambda/latest/dg/with-on-demand-https.html and https://docs.aws.amazon.com/lambda/latest/dg/lambda-services.html#intro-core-components-event-sources).

Reply:
I think it makes sense to keep this language. In most cases there will be no http span before the faas span anyways, and in the rare (nonexistent?) case that there are two nested spans, it is ok I think to duplicate some information.

Reply:
I think adding this requirements could cause headaches. E.g. if a sensors only offers an enable/disable API but no "add attributes from *this* request object to *this* span" API, then implementing this will be hard. Also, this would mean that faas depends on HTTP, messaging, ... sensors.

Reply:
I share your concerns, however I think this requirement should be in. Without such attributes, a possible correlation on the source that triggered the lambda is not possible. 
My opinion is also given by the fact that lambda is anyway a special use case for OpenTelemetry and with such, the faas sensor should be handle as a special case.

Reply:
How does this add a dependency on another sensor? In most cases, the faas sensor will be the local root. it will just look in the request object, notice it is an http style call, and add the http attributes in addition to the faas attributes. I would expect the http sensor not to even create an incoming span most of the time.

Reply:
Ah, so you mean since HTTP-style faas invocations won't go through a normal HTTP code path, a normal HTTP sensors is not required since the faas needs to re-implement a faas HTTP sensor anyway. Makes sense! 👍  Having understood that, I'm fine with this requirement. 😃

Reply:
> I think it makes sense to keep this language. In most cases there will be no http span before the faas span anyways, and in the rare (nonexistent?) case that there are two nested spans, it is ok I think to duplicate some information

Having a root span just for FaaS may cause issues in e.g. AWS SQS where a bulk of messages is passed to the function. Each message may contain it's own tag and therefore we should end up in one Span per message correctly linked to the remote sender. If there is a root span these message spans would use this one as parent and the link to the senders may be lost.

Main question here is what should our service detection show? One service per function invocation or more concrete what the function is used for.

Reply:
In that case would the sensor just not create a `faas` root span and instead create one `faas` root span for each message, using the `messaging` semconv?

Reply:
It depends on the implementation - which is done based on the spec.

Reply:
I agree with @Flarna that we need to specify better each case. At the moment, we don't have a semconv for messaging queue (in the otel-spec repository). I suggest to add a TODO and solve it based on the feedback received from when we PR https://github.com/dynatrace-oss-contrib/opentelemetry-specification/pull/1